components/postgresql/postgresql.app.mjs

```diff
@@ -1,11 +1,193 @@
+      return rows.map((row) => row.column_name);
+    },
+    /**
+     * Gets a table's primary key field
+     * @param table {string} Name of the table to get the primary key for
+     * @returns Name of the primary key column
+     */
+    async getPrimaryKey(table) {
+      const rows = await this.executeQuery(`SELECT a.attname FROM pg_index i JOIN pg_attribute a ON a.attrelid = i.indrelid AND a.attnum = ANY(i.indkey) WHERE i.indrelid = '${table}'::regclass AND i.indisprimary`);
+      return rows[0].attname;
+    },
+    /**
+     * Gets rows in a table with values greater than lastResult
+     * @param table {string} Name of database table to query
+     * @param column {string} Column to filter by lastResult
+     * @param lastResult {string} A column value. Used to retrieve new rows since the
+     *  last time the table was queried.
+     * @returns Array of rows returned from the query
+     */
+    async getRows(table, column, lastResult = null) {
+      const query = `SELECT * FROM ${table} ${lastResult
+        ? "WHERE " + column + ">" + lastResult
+        : ""} ORDER BY ${column} DESC`;
+      const { rows } = await this.executeQuery(query);
+      return rows;
+    },
+    /**
+     * Gets a single row from a table based on a lookup column & value
+     * @param table {string} Name of database table to query
+     * @param column {string} Column to filter by value
+     * @param value {string} A column value to search for
+     * @returns A single database row
+     */
+    async findRowByValue(table, column, value) {
+      const rows = await this.executeQuery(`SELECT * FROM ${table} WHERE ${column} = '${value}'`);
+      return rows[0];
+    },
+    /**
+     * Gets the number of rows in a table
+     * @param table {string} Name of database table to query
+     * @returns Count of how many rows exist in the table
+     */
+    async getRowCount(table) {
+      const response = await this.executeQuery(`SELECT COUNT(*) FROM ${table}`);
+      return response[0].count;
+    },
+    /**
+     * Deletes a row or rows in a table based on a lookup column & value
+     * @param table {string} Name of database table to delete from
+     * @param column {string} Column used to find the row(s) to delete
+     * @param value {string} A column value. Used to find the row(s) to delete
+     */
+    async deleteRows(table, column, value) {
+      await this.executeQuery(`DELETE FROM ${table} WHERE ${column} = '${value}'`);
+    },
+    /**
+     * Inserts a row in a table
+     * @param table {string} Name of database table to insert row into
+     * @param columns {array} Array of column names
+     * @param values {array} Array of values corresponding to the column names provided
+     * @returns The newly created row
+     */
+    async insertRow(table, columns, values) {
+      const { rows } = this.executeQuery(`INSERT INTO ${table} (${columns.join()}) VALUES (${values.join()}) RETURNING *`);
```

Comment:
You are missing the `await` here. Also you can remove the `.join()` function from the template literal since the array will be converted automatically into an string of names separated by comas.

Also there is a caveat on the sql query syntax here, because it seems like if I'm sending a column name that was declared as a VARCHAR it throws an error. So my suggestion is to map the values and force them to be enclosed by single quotes like this
```javascript
${values.map((v) => "'" + v + "'")
```
Otherwise you need to advice the user in the property description that he needs to set those values inside `''`